From e62c15827e22892467c09ef355aece2f0257b6ad Mon Sep 17 00:00:00 2001
Message-Id: <e62c15827e22892467c09ef355aece2f0257b6ad.1427265330.git.chang-joon.lee@intel.com>
In-Reply-To: <b4c66aa6a04d2be0fca9fa4448c7188a581a57a5.1427265330.git.chang-joon.lee@intel.com>
References: <b4c66aa6a04d2be0fca9fa4448c7188a581a57a5.1427265330.git.chang-joon.lee@intel.com>
From: Durgadoss R <durgadoss.r@intel.com>
Date: Thu, 5 Mar 2015 17:34:51 +0530
Subject: [PATCH 15/22] FOR_UPSTREAM [VPG]: drm/i915: gracefully handle
 unsupported pixel format

Whenever the user space is sending a flip with invalid pixel
format, we raise a BUG() in i9xx_update_primary_plane().
This panics the kernel unnecessarily. So, handle this
greacefully with an appropriate error message, and fail
this particular flip.

Issue: GMINL-7183
Change-Id: I6e89201879c26b47dfc75e228416a1a3b8e9be8c
Signed-off-by: Durgadoss R <durgadoss.r@intel.com>
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 24009e0..25f177f 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -2765,7 +2765,9 @@ static void i9xx_update_primary_plane(struct drm_crtc *crtc,
 			dspcntr |= DISPPLANE_RGBA101010;
 		break;
 	default:
-		BUG();
+		DRM_ERROR("Invalid pixel format: %s\n",
+				drm_get_format_name(fb->pixel_format));
+		return;
 	}
 
 	if (intel_crtc->rotate180)
-- 
1.7.9.5

